test/integration/service/loadbalancer_test.go

```diff
@@ -284,21 +270,16 @@ func Test_ServiceLoadBalancerEnableLoadBalancerClass(t *testing.T) {
 		},
 	}
 
-	_, err = client.CoreV1().Services(ns.Name).Create(ctx, service, metav1.CreateOptions{})
+	_, err = client.CoreV1().Services(ns.Name).Create(context.TODO(), service, metav1.CreateOptions{})
 	if err != nil {
 		t.Fatalf("Error creating test service: %v", err)
 	}
-
-	if len(cloud.Calls) > 0 {
```

Comment:
Why is this check being removed?

Reply:
We do expect the cloud call to be 0, but when I review the test output from prow, I realize that the controller runs in an async way, so we can't ensure when the code running into this line the `ensureLoadBalancer` function gets called, so I don't think it makes sense to add this check here.
The controller runs using via this:
```go
go controller.Run(ctx, 1)
```
Do you have any idea that we can run this in order? Thanks!

Reply:
It's not pretty but you can add a delay with time.Sleep to let the controller reconcile state first. It's unforunately very common pattern when testing controllers because they reconcile things async.

Reply:
Gotcha, will do that!